Pin command fixes

Stebalien opened the pull request:
1. `ipfs pin ls` now defaults to "all", not "recursive".
2. `ipfs pin rm` should respond with `Unpinned: HASH`, not `Pinned: HASH`. This is a breaking change.

I don't know how you feel about multiple commits per pull but these were both trivial.

whyrusleeping commented:
Maybe instead of creating a new object for the unpinning, we should change the field name of the existing object to something agnostic like 'Pins' or 'Refs'

Stebalien commented:
How about Hash (to be consistent with the add command)?

On March 15, 2016 5:23:01 PM EDT, Jeromy Johnson <notifications@github.com> wrote:
>Maybe instead of creating a new object for the unpinning, we should
>change the field name of the existing object to something agnostic like
>'Pins' or 'Refs'
>
>---
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/ipfs/go-ipfs/pull/2477#issuecomment-197025627

-- 
Steven Allen

dignifiedquire commented:
I don't think making `ipfs pin ls` default to `all` is a good idea. The reason being that `all` is easily very, very large and often much less useful because of that.

Stebalien commented:
@whyrusleeping,

> Maybe instead of creating a new object for the unpinning, we should change the field name of the existing object to something agnostic like 'Pins' or 'Refs'

I didn't realize this was a list. I changed it to Pins.

@dignifiedquire 

That change was made in 28fe2e4f4cb837a463fa912e7054cf8a95f27800. I just updated the documentation.

whyrusleeping commented on core/commands/pin.go:
```diff
@@ -187,7 +187,7 @@ Example:
 	},
 	Options: []cmds.Option{
-		cmds.StringOption("type", "t", "The type of pinned keys to list. Can be \"direct\", \"indirect\", \"recursive\", or \"all\". Defaults to \"recursive\"."),
+		cmds.StringOption("type", "t", "The type of pinned keys to list. Can be \"direct\", \"indirect\", \"recursive\", or \"all\". Defaults to \"all\"."),
```
Could we remove the 'Defaults to "All"' from the help string and add `.Default("all")` to the end of the `StringOption` constructor?

The down later where we check `if typeStrFound {` we can drop that check and just rely on the code to pick its own default value correctly.

whyrusleeping replied:
an example: https://github.com/ipfs/go-ipfs/blob/master/core/commands/pin.go#L44

RichardLitt commented:
@whyrusleeping which change? default to all?

whyrusleeping commented:
@RichardLitt relevant to the API:

the json object sent by the pin add and rm commands now has a field of simply `Pins` instead of `Pinned`.

It also corrects the documentation to say that "all" is the default instead of "recursive" (no code changes were made for this, all was already the default)

RichardLitt commented:
@whyrusleeping Cool. Will update http api when this is merged.

whyrusleeping commented:
This looks good to me, @Stebalien could you rebase the changes on latest master?

Stebalien commented:
@whyrusleeping Done.

chriscool commented:
@Stebalien the change to default to "all" was made following this discussion: https://github.com/ipfs/go-ipfs/pull/2208#discussion_r50091114

To summarize the discussion, the reason is that I added arguments to `ipfs pin ls` so that one can easily check if a given hash is pinned or not. And when you do `ipfs pin ls HASH` it is natural to default to "all", so for consistency Juan and myself decided that "all" should always be the default.

RichardLitt commented:
Thanks @chriscool.
